evm/src/main/java/org/hyperledger/besu/evm/operation/BlockHashOperation.java

```diff
@@ -41,10 +43,10 @@ public BlockHashOperation(final GasCalculator gasCalculator) {
   @Override
   public Operation.OperationResult executeFixedCostOperation(
       final MessageFrame frame, final EVM evm) {
-    final UInt256 blockArg = UInt256.fromBytes(frame.popStackItem());
+    final Bytes blockArg = frame.popStackItem().trimLeadingZeros();
```

Comment:
might this lead to problems if we are not aligned on a word boundary?

Reply:
It won't.  The call to `blockArg.toLong()` on line 54 handles any byte length of 8 or less. That's the only use of this variable.